standard_rules/unusual_login.py

```diff
@@ -96,18 +101,52 @@ def rule(event):
     # Mocked during unit testing
     put_string_set(event_key, [json.dumps(updated_geo_logins)])
 
+    global GEO_HISTORY  # pylint: disable=global-statement
+    GEO_HISTORY = updated_geo_logins
+    logging.debug("GEO_HISTORY in main rule:\n%s", json.dumps(GEO_HISTORY))
     return True
 
 
-def get_key(event):
+def get_key(event) -> str:
     # Use the name to deconflict with other rules that may also use actor_user
     return __name__ + ":" + str(event.udm("actor_user"))
 
 
 def title(event):
+    event_time_truncated = nano_to_micro(event.get("p_event_time"))
+    parse_time_truncated = nano_to_micro(event.get("p_parse_time"))
+    time_delta = resolve_timestamp_string(parse_time_truncated) - resolve_timestamp_string(
+        event_time_truncated
+    )
+    days = time_delta.days
+    hours, remainder = divmod(time_delta.seconds, 3600)
+    minutes, seconds = divmod(remainder, 60)
+    parsing_delay = ""
+    if days > 0:
+        parsing_delay = f"{days} day(s) "
+    if hours > 0:
+        parsing_delay = "".join([parsing_delay, f"{hours} hour(s) "])
+    if minutes > 0:
+        parsing_delay = "".join([parsing_delay, f"{minutes} minute(s) "])
+    if seconds > 0:
+        parsing_delay = "".join([parsing_delay, f"{seconds} second(s)"])
```

Comment:
I feel like this is generally useful logic! Can we put this in a helper function (something like `friendly_duration`) that takes whatever type `time_delta` is and returns a string? Then this becomes:

```python
from panther_base_helpers import  friendly_duration
parsing_delay  =  friendly_duration(time_delta)
```
and in `panther_base_helpers`:
```python
def friendly_duration(time_delta) -> str:
    days = time_delta.days
    hours, remainder = divmod(time_delta.seconds, 3600)
    minutes, seconds = divmod(remainder, 60)
    parsing_delay = ""
    if days > 0:
        parsing_delay = f"{days} day(s) "
    if hours > 0:
        parsing_delay = "".join([parsing_delay, f"{hours} hour(s) "])
    if minutes > 0:
        parsing_delay = "".join([parsing_delay, f"{minutes} minute(s) "])
    if seconds > 0:
        parsing_delay = "".join([parsing_delay, f"{seconds} second(s)"])
    return parsing_delay
```

Reply:
I guess i could also use "alert time" instead of "parsing time", and that would actually get closer to my intent of highlighting the delta between the alert and the event.